modules/tinymce/src/core/test/ts/browser/newline/ForcedRootBlockTest.ts

```diff
@@ -0,0 +1,163 @@
+import { ApproxStructure, GeneralSteps, Logger, Pipeline, Step } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader } from '@ephox/mcagar';
+import { Obj } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+import InsertNewLine from 'tinymce/core/newline/InsertNewLine';
+import Theme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.core.newline.ForcedRootBlockTest', (success, failure) => {
+  Theme();
+
+  const bookmarkSpan = '<span data-mce-type="bookmark" id="mce_2_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px"></span>';
+  const forcedRootBlock = 'p';
+  const forcedRootBlockAttrs = { style: 'color: red;', class: 'abc def' };
+
+  const baseExpectedHTML = (innerHTML: string) => `<p class="${forcedRootBlockAttrs.class}" style="${forcedRootBlockAttrs.style}">${innerHTML}</p>`;
+
+  const sInsertNewline = (editor: Editor, args) => {
+    return Step.sync(function () {
+      InsertNewLine.insert(editor, args);
```

Comment:
This is sort of bypassing a lot of editor logic, it'd be better to do this via the `mceInsertNewLine` command. So this could just become `tinyApis.sExecCommand('mceInsertNewLine')`.

- [ ] Use execCommand instead of importing internal modules